### DISCUSSION TITLE
INdexing stuck in middle

### DISCUSSION THREAD

**Discussion by kalpeshsomani007** [oracle/opengrok#4204] [General]
It always stuck on below eroor 
 
Could not notify the webapp that project device:indexed=true,history=true was indexed: InboundJaxrsResponse{context=ClientResponse{method=PUT, uri=http://localhost:9999/source/api/v1/projects/device/indexed, status=404, reason=Not Found}}
below logs files attached

[opengrok0.0.log.zip](https://github.com/oracle/opengrok/files/10926660/opengrok0.0.log.zip)

**Comment by vladak** [comment]
You need to deploy the webapp first to the URL. HTTP 404 error means there is nothing on that URL.

**Reply by kalpeshsomani007** [reply]
@vladak  webapp already deployed

**Reply by kalpeshsomani007** [reply]
Webapps is deployed
![image](https://user-images.githubusercontent.com/25900639/224227996-0fdff4da-d279-4525-9602-9983b875aded.png)

![image](https://user-images.githubusercontent.com/25900639/224228099-d8a74756-f9f2-485f-a1c1-fb202fbd2966.png)
![image](https://user-images.githubusercontent.com/25900639/224228127-cbfb046f-cabf-4180-8a6b-a7142c850aa9.png)


I have run the below command 
opengrok-indexer     -J=-Djava.util.logging.config.file=etc/logging.properties     -a dist/lib/opengrok-1.8.2.jar --     -c /usr/local/bin/ctags     -s src/aosp-caf/ -d data -H -P -S -G     -W etc/configuration.xml -U http://localhost:9999/source

Most of the time it stuck on below line
 WARNING: Failed to set last modified time on '/home/rshaik/Documents/workspace/opengrok/data/timestamp', used for timestamping the index database.
[opengrok0.0.log.zip](https://github.com/oracle/opengrok/files/10938704/opengrok0.0.log.zip)

**Reply by vladak** [reply]
If the webapp was deployed successfully, you would not be getting the HTTP 404 error when trying to access the URL on which it should be listening for requests (http://localhost:9999/source/ in your case). The above screenshot merely says that the application server is up, however does not say anything about deployed web applications, specifically the OpenGrok web app.

In my case, Tomcat is listening on port 8080 and when I access http://localhost:8080/manager/html (after properly configuring the admin user in Tomcat's `tomcat-users.xml` configuration and restarting it), it shows this:

![Screenshot from 2023-03-10 12-33-25](https://user-images.githubusercontent.com/2934284/224306708-61ee3ec3-80b9-4a59-8fae-1891e7c7dcd6.png)

As visible in the screenshot, in my case, the OpenGrok web app is deployed to URL http://localhost:8080/opengrok_web_war/

**Reply by vladak** [reply]
Anyhow, let me explain why the web app needs to be accessible for the indexer: when the indexer is running, it needs to change the state of the of the webapp, in particular the configuration generated during indexing needs to be set in the web application so that it knows what projects/repositories are present. These changes happen via by sending API requests to the web app URL. Also, during indexing, the indexer sends some API requests to the webapp to mark projects as indexed and also to refresh index searchers in the web app so that search results can reflect the latest state of the input data.

Now, it is perfectly fine to run the indexer without the web app being accessible, just do not use the -U indexer option. It really depends on what is your goal of the indexing.

**Comment by kalpeshsomani007** [comment]
Do i need to change the command ??
opengrok-indexer -J=-Djava.util.logging.config.file=etc/logging.properties -a dist/lib/opengrok-1.8.2.jar -- -c /usr/local/bin/ctags -s src/aosp-caf/ -d data -H -P -S -G -W etc/configuration.xml -U http://localhost:9999/source

THIS COMMAND WORKS PERFECTLY WHEN WE HAVE SHORT DIRECTORy. bUT iF WE HAVE LARGE DIRECTORY IT'S GETTING STUCK MOST OF THE TIME

LET'S SAY IN ABOVE LOGS INDEXING COMPLETED 28 DIRECTORY OUT OF 34. YOU CAN SEE FROM THE LOGS
SINGLE DIRECTORY IT'S DEPLOYING PERFECTLY.

CONFIGURATION WHICH I USE
TOMCAT -10
OPENGROK -1.8.2
CTAGE - UNIVERSAL CTAG

**Reply by vladak** [reply]
Huh, is your caps-lock key stuck or something ? Anyhow, the indexer options look good to me. Of course, it depends on what is the intention, in particular w.r.t. the -U option.

If the indexer seems to be stuck, then my advice would be to raise the log level (see https://github.com/oracle/opengrok/wiki/Debugging#web-application) and see what exactly is going on. Maybe also use the `--progress` indexer option.

**Reply by kalpeshsomani007** [reply]
yes caps lock stuck sorry for that.

I want to deploy the Whole android code to opengrok but in the middle of indexing it is getting stuck. 

is there any command where if the file is taking time it should skip after some time & go on next file ??

**Reply by vladak** [reply]
See my previous pieces of advice.
